jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -1826,7 +1842,7 @@ public void setMetaData(org.eclipse.jetty.http.MetaData.Request request)
         }
         else if (encoded.startsWith("/"))
         {
-            path = (encoded.length() == 1) ? "/" : URIUtil.canonicalPath(uri.getDecodedPath());
+            path = (encoded.length() == 1) ? "/" : uri.getDecodedPath();
```

Comment:
Few lines below:

```java
if (path == null || path.isEmpty())
{
    setPathInfo(encoded == null ? "" : encoded);
    throw new BadMessageException(400, "Bad URI");
}
```

Seems to me that `path` could be legitimately null at that point, and if not, why do we set the pathInfo, just to throw the next line?

Reply:
I don't think we can continue with a null path.  However, we still set the path info before throwing so we don't break any error page handling and/or request logging that might get confused by a null or empty request path.

Reply:
Also remember this is cleaned up a lot in 10, so I don't want to fix too many unrelated issues here.

Reply:
Okay then.